pl_bolts/datamodules/emnist_datamodule.py

```diff
@@ -0,0 +1,221 @@
+            )
+
+        if split not in self.dataset_cls.splits:
+            raise ValueError(
+                f"Unknown value '{split}' for argument `split`. Valid values are {self.dataset_cls.splits}."
+            )
+
+        super(EMNISTDataModule, self).__init__(  # type: ignore[misc]
+            data_dir=data_dir,
+            val_split=val_split,
+            num_workers=num_workers,
+            normalize=normalize,
+            batch_size=batch_size,
+            seed=seed,
+            shuffle=shuffle,
+            pin_memory=pin_memory,
+            drop_last=drop_last,
+            *args,
+            **kwargs,
+        )
+        self.split = split
+
+        if strict_val_split:
+            # replaces the value in `val_split` with the one defined in the paper
+            if self.split in self._official_val_split:
+                self.val_split = self._official_val_split[self.split]
+            else:
+                raise ValueError(
+                    f"Invalid value '{self.split}' for argument `split` with `strict_val_split=True`. "
+                    f"Valid values are {set(self._official_val_split)}."
+                )
+
+    @property
+    def num_classes(self) -> int:
+        """Returns the number of classes. See the table above."""
+        return len(self.dataset_cls.classes_split_dict[self.split])
+
+    def prepare_data(self, *args: Any, **kwargs: Any) -> None:
+        """
+        Saves files to ``data_dir``.
+        """
+
+        self.dataset_cls(self.data_dir, split=self.split, train=True, download=True)
+        self.dataset_cls(self.data_dir, split=self.split, train=False, download=True)
+
+    def setup(self, stage: Optional[str] = None) -> None:
+        """
+        Creates train, val, and test dataset
+        """
+
+        if stage == "fit" or stage is None:
+            train_transforms = self.default_transforms() if self.train_transforms is None else self.train_transforms
+            val_transforms = self.default_transforms() if self.val_transforms is None else self.val_transforms
+
+            dataset_train = self.dataset_cls(
+                self.data_dir, split=self.split, train=True, transform=train_transforms, **self.EXTRA_ARGS
+            )
+            dataset_val = self.dataset_cls(
+                self.data_dir, split=self.split, train=True, transform=val_transforms, **self.EXTRA_ARGS
+            )
+
+            # Split
+            self.dataset_train = self._split_dataset(dataset_train)
+            self.dataset_val = self._split_dataset(dataset_val, train=False)
+
+        if stage == "test" or stage is None:
+            test_transforms = self.default_transforms() if self.test_transforms is None else self.test_transforms
+            self.dataset_test = self.dataset_cls(
+                self.data_dir, split=self.split, train=False, transform=test_transforms, **self.EXTRA_ARGS
+            )
+
+    def default_transforms(self) -> Callable:
```

Comment:
This can be simplified

Reply:
Should be addressed now. Would you mind having a look again?